content/docs/concepts/query-rules.md

```diff
@@ -0,0 +1,177 @@
+***If*** **search query contains 'Google'**, where the `operator` is `contains` and `query` is `Google`.
+
+<!-- If condition image -->
+![](https://i.imgur.com/j4qodaD.png)
+
+The second step consists of configuring of actions that are triggered when the ***If*** condition of the query is 
+matched. Query Rule currently supports two `actions`:
+
+- `promote`: appends external docs or results along with the actual search results
+- `hide`: prevents certain results from getting included in the resturned search results
+
+<!-- Then condition image each for promote and hide -->
+![](https://i.imgur.com/biuc4SL.png)
+
+## Part 3: Query Rules using API
+
+The dashboard provides an easier way to tune and visualize the query rules, however you can also manage them via API.
+The rule can be divided into two main segments:
+
+1. ***If*** clause:
+
+The **If** clause comprises of both `query` and `operator`. Currently supported `operators` are listed in the 
+section above. Whenever a search query matches the **If** clause, it automatically triggers the **Then** clause of the
+rule.
+
+
+```js
+{
+    ..
+    "if": {
+        "query": "harry potter",
+        "operator": "contains"
+    } 
+    ...
+}
+```
+
+2. ***Then*** clause:
+
+The **Then** clause comprises of `actions` that are to be executed when a rule is triggered. Currently supported 
+`actions` are listed in the section above.
+
+```js
+{
+    ...
+    "then": {
+        "promote": [
+            {
+                "id": "harry_potter_cheat_sheet",
+                "name": "Harry Potter",
+                "section_order": [
+                    "Books",
+                    "Movies",
+                    "Franchise"
+                ],
+                "template_type": "reference"
+            }
+        ],
+        "hide": [
+            { "doc_id": "Jle44WgBnfYvZBcA0H66" }
+        ]
+    }
+    ...
+}
+```
+
+A complete query rule may look similar to:
+
+```js
+{
+    "id": "contains_harry_potter",
+    "if": {
+        "query": "harry potter",
+        "operator": "contains"
+    },
+    "then": {
+        "promote": [
+            {
+                "id": "harry_potter_cheat_sheet",
+                "name": "Harry Potter",
+                "section_order": [
+                    "Books",
+                    "Movies",
+                    "Franchise"
+                ],
+                "template_type": "reference"
+            }
+        ],
+        "hide": [
+            { "doc_id": "Jle44WgBnfYvZBcA0H66" }
+        ]
+    }
+}
+```
+
+**Note**: The rule `id` is an optional field, in case it isn't provided, it will be generated automatically 
+based on the `query` and `operator`.
+
+Now that we have constructed the `rule` object, we can create the query rule by executing the following request:
+
+```sh
+curl --location --request POST "https://accapi.appbase.io/app/book-store/rule" \
+  --header "Content-Type: application/json" \
+  --data "
+{
```

Comment:
@utsavoza Postman creates escaped API request body, but we can use a single quote outside and then unescape the values, makes it easier to read.